satellite/satellitedb/metainfo.sql

```diff
@@ -0,0 +1,116 @@
+CREATE TYPE encryption_parameters AS (
+    -- total 5 bytes
+    ciphersuite BYTE NOT NULL;
+    block_size  INT4 NOT NULL;
+);
+
+CREATE TYPE redundancy_scheme AS (
+    -- total 9 bytes
+    algorithm   BYTE   NOT NULL;
+    share_size  INT4   NOT NULL;
+    required    INT2   NOT NULL;
+    repair      INT2   NOT NULL;
+    optimal     INT2   NOT NULL;
+    total       INT2   NOT NULL;
+);
+
+CREATE TABLE buckets (
+    bucket_id      UUID; -- 16 bytes, should we use here a serial of 8 bytes?
+    project_id     UUID; -- 16 bytes
+
+    bucket_name BYTEA NOT NULL;     -- ? bytes
+
+    created_at     TIMESTAMP NOT NULL; -- 8 bytes
+    modified_at    TIMESTAMP NOT NULL; -- 8 bytes
+    attribution_id UUID;               -- 16 bytes (or should this be a serial of 8 bytes, should this be on object?)
+
+    path_encryption_algorithm BYTE NOT NULL; -- 1 byte
+
+    default_segment_size INT4 NOT NULL;
+
+    default_encryption encryption_parameters NOT NULL; -- 5 bytes
+    default_redundancy redundancy_scheme     NOT NULL; -- 9 bytes
+
+    PRIMARY KEY (bucket_id, project_id);
+)
+
+CREATE TYPE object_status AS ENUM ('partial', 'committing', 'committed', 'deleting');
+
+CREATE TABLE objects (
+    bucket_id      UUID  NOT NULL REFERENCES buckets(bucket_id); -- 16 bytes (or 8 bytes when serial)
+    encrypted_path BYTEA NOT NULL; -- ?  bytes
+    stream_id      UUID  NOT NULL; -- 16 bytes (or should this be a serial of 8 bytes)
+
+    status         object_status  NOT NULL DEFAULT 'partial'; -- 1 byte
+    version        INT4           NOT NULL DEFAULT 0;         -- 4 bytes
+
+    created_at          TIMESTAMP NOT NULL; -- 8 bytes
+    expires_at          TIMESTAMP NOT NULL; -- 8 bytes
+    encrypted_metadata  BYTEA     NOT NULL; -- ? bytes
+
+    data_checksum      INT8       NOT NULL DEFAULT -1; -- 8 bytes, checksum of checksums (do we need this)
+    total_size         INT8       NOT NULL DEFAULT -1; -- 8 bytes
+    fixed_segment_size INT4       NOT NULL DEFAULT -1; -- 4 bytes
+    segment_count      INT4       NOT NULL DEFAULT -1; -- 8 bytes (do we need this or can we use segments table as source of truth)
+
+    encryption encryption_parameters NOT NULL; -- 5 bytes (should this be a 4 byte reference to a table)
+    redundancy redundancy_scheme     NOT NULL; -- 9 bytes (should this be a 4 byte reference to a table)
+
+    PRIMARY KEY (bucket_id, encrypted_path, version);
+)
+
+CREATE TABLE segments (

```

Comment:
Where we will keep current `RemotePieces` and do we plan to store piece hash at some point?

Reply:
Ok, I see answer to my first question. Its in `inline_data_or_nodes`.